lib/quayio/scanner/image.rb

```diff
@@ -44,14 +45,22 @@ def tag
       def raw_image
         return @raw_image if defined? @raw_image
 
-        @raw_image = begin
-          JSON.parse(
-            RestClient.get("https://quay.io/api/v1/repository/#{repo}/tag/#{tag}/images",
-                           authorization: "Bearer #{quayio_token}", accept: :json)
-          )['images'].first
-        rescue RestClient::ExceptionWithResponse => err
-          return nil if err.http_code == 404 # ignore unknown repos
-          raise err
+        (1..MAX_ATTEMPTS).each do |attempt|
+          begin
+            response = RestClient.get(
+              "https://quay.io/api/v1/repository/#{repo}/tag/#{tag}/images",
+              authorization: "Bearer #{quayio_token}",
+              accept: :json)
+          rescue RestClient::ExceptionWithResponse => err
+            return nil if err.http_code == 404 # ignore unknown repos
+            if err.http_code == 520 and attempt < MAX_ATTEMPTS
+              sleep(rand(10))
```

Comment:
Warum hier ein Zufall beim warten auf den nächsten Versuch? Ich kenne sonst das Pattern, dass mit jedem weiteren Versuch länger gewartet wird (als Optimierung der Gesamtwartezeit). Ich würde daher schreiben

```suggestion
              sleep(attempt ** 2)
```

Das `**` steht für Potenz.

Reply:
Das Problem scheint daher zu kommen, dass bei einem Check-Request alle Docker-Nodes gleichzeitig zu quay.io rennen. Das würde sich nicht ändern, wenn alle Nodes deterministisch warten, um dann erneut gleichzeitig bei quay.io anzufragen. Ich will hier bewusst für eine zufällige zeitliche Streuung der Requests über die einzelnen Nodes sorgen.